robot/impl/robot_framesystem_test.go

```diff
@@ -0,0 +1,245 @@
+package robotimpl
+
+import (
+	"context"
+	"math"
+	"testing"
+	"time"
+
+	"github.com/golang/geo/r3"
+	"go.viam.com/test"
+	"go.viam.com/utils/testutils"
+
+	"go.viam.com/rdk/components/base"
+	"go.viam.com/rdk/components/gripper"
+	"go.viam.com/rdk/config"
+	"go.viam.com/rdk/logging"
+	"go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/services/datamanager"
+	_ "go.viam.com/rdk/services/datamanager/builtin"
+	"go.viam.com/rdk/services/motion"
+	_ "go.viam.com/rdk/services/motion/builtin"
+	"go.viam.com/rdk/services/sensors"
+	_ "go.viam.com/rdk/services/sensors/builtin"
+	"go.viam.com/rdk/spatialmath"
+	rdktestutils "go.viam.com/rdk/testutils"
+	"go.viam.com/rdk/testutils/robottestutils"
+	rutils "go.viam.com/rdk/utils"
+)
+
+func TestFrameSystemConfigWithRemote(t *testing.T) {
```

Comment:
I like this.  This is a better place for these tests since there should be no concept of remotes in the framesystem package

Reply:
there's an argument that all the framesystem tests that create a local robot should be in here instead, but did not do that in this pr

Reply:
Thats a good point.  Looking through the tests this would be a massive pain though, since right now we're relying on a robot level helper function to actually extract the framesystem from a given config